includes/Core/Modules/Modules.php

```diff
@@ -804,13 +804,17 @@ private function update_module_settings( Module $module, array $data ) {
 	 * @return array Module REST response data.
 	 */
 	private function prepare_module_data_for_response( Module $module ) {
+		$module_info = $module->prepare_info_for_js();
+
 		return array(
 			'slug'         => $module->slug,
-			'name'         => $module->name,
-			'description'  => $module->description,
-			'homepage'     => $module->homepage,
-			'internal'     => $module->internal,
-			'order'        => $module->order,
+			'name'         => isset( $module_info['name'] ) ? $module_info['name'] : $module->name,
+			'description'  => isset( $module_info['description'] ) ? $module_info['description'] : $module->description,
+			'homepage'     => isset( $module_info['homepage'] ) ? $module_info['homepage'] : $module->homepage,
+			'internal'     => isset( $module_info['internal'] ) && $module_info['internal'],
+			'order'        => isset( $module_info['sort'] ) ? $module_info['sort'] : $module->order,
+			'provides'     => isset( $module_info['provides'] ) ? $module_info['provides'] : array(),
+			'autoActivate' => isset( $module_info['autoActivate'] ) && $module_info['autoActivate'],
```

Comment:
Oh now I see you already changed this. Generally +1, however:
* `autoActivate` should become `forceActive` and rely on `$module->force_active`.
* `provides` should not be part of this, let's handle that one purely on the client. In the long term it should be removed from the server.

Also, the new property needs to be covered in the `get_module_schema()` method too.

Reply:
Ok, updated.